### PR TITLE
LPS-119924 We don't need to save on blur anymore

### PR DESCRIPTION
**Pull request by sandrodw3**

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js [hunk 1]
```diff
@@ -132,14 +136,12 @@ export default function getAlloyEditorProcessor(
 					) {
 						event.cancel();
 					}
+
+					debouncedChangeCallback();
 				}),
 
 				nativeEditor.on('blur', () => {
 					if (_editor._mainUI.state.hidden) {
-						if (_callbacks.changeCallback) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js:139]: I am checking this `_callbacks` variable and I don't think we need it at all :laughing:

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js [hunk 2]
```diff
@@ -121,6 +121,10 @@ export default function getAlloyEditorProcessor(
 
 			const nativeEditor = _editor.get('nativeEditor');
 
+			const debouncedChangeCallback = debounce(() => {
+				changeCallback(nativeEditor.getData());
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js:125]: This `changeCallback` might be undefined, we need to check

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js (hunk 1 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js:139]: 🐸

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js (hunk 2 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/processors/getAlloyEditorProcessor.js:125]: 🧞‍♂️

**Comment by sandrodw3** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
Leaving this on hold to see if @victorg1991 remembers if we hade some motive to don't do this in the past 🧀

**Comment by p2kmgcl** [comment]
:cheese:

**Comment by sandrodw3** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
ci:test:sf

**Comment by sandrodw3** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
Hey @Tim-Cao, we would appreciate your help here. As the same test has failed on both relevant runs, I think it's related. We have modified the alloy editor to save while typing instead of on blur, and I think it has affected that undo/redo case. Could you take a look? 🙂

**Comment by victorg1991** [comment]
👏

**Comment by sandrodw3** [comment]
ci:forward

**Pull request closed without merging**